### PR TITLE
Use the filters in the backend to select the right forms

### PR DESCRIPTION
**Pull request by sifelix**
image

### CODE REVIEW

File: packages/interfaces/src/kobo/GetFormsDto.ts [hunk 1]
```diff
@@ -1,5 +1,10 @@
-import { IsString } from 'class-validator';
+import { IsDateString, IsOptional, IsString } from 'class-validator';
 
 export class GetFormsDto {
-  @IsString() readonly DisTyp!: string;
+  @IsString() readonly disTyp!: string;
+  @IsDateString() readonly startDate!: string;
+  @IsDateString() readonly endDate!: string;
+  @IsOptional() @IsString() readonly province?: string;
```
Comment (p-chavoshi) [inline packages/interfaces/src/kobo/GetFormsDto.ts:7]: this parameter should be optional?

File: apps/frontend/services/api/kobo/useGetForms.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
+import { DroughtDto, FloodDto, IncidentDto } from '@wfp-dmp/interfaces';
+import useSWR from 'swr';
+
+import { SearchFormData } from 'components/Filters/SearchFilters';
```
Comment (p-chavoshi) [inline apps/frontend/services/api/kobo/useGetForms.ts:4]: a little weird to export a type from inside a component no?

File: apps/backend/src/modules/kobo/tests/__mocks__/getForms.ts [hunk 3]
```diff
@@ -8,7 +9,21 @@ import { droughtMock } from './drought';
...
+export const getFormsMock = ({
+  disTyp,
+  province,
+  district,
+  commune,
+  startDate,
+  endDate,
+}: {
+  disTyp: string;
+  province: string | undefined;
```
Comment (p-chavoshi) [inline apps/backend/src/modules/kobo/tests/__mocks__/getForms.ts:21]: why undefined?

**Review by p-chavoshi — APPROVED** [review]

File: apps/frontend/services/api/kobo/useGetForms.ts (hunk 2 above)
Comment (sifelix) [reply apps/frontend/services/api/kobo/useGetForms.ts:4]: It is consistent with the other filters but agreed let's move then to a specific file in a another ticket

**Pull request merged**